src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -18,7 +18,7 @@ namespace Arcus.Messaging.Pumps.ServiceBus
     ///     Message pump for processing messages on an Azure Service Bus entity
     /// </summary>
     /// <typeparam name="TMessage">Type of expected message payload</typeparam>
-    public abstract class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
+    public class AzureServiceBusMessagePump<TMessage> : MessagePump<TMessage, AzureServiceBusMessageContext>
```

Comment:
`TMessage` should no longer be required since that's what the `IMessageHandler` is for.

`MessagePump` should create an instance of `IMessageHandler` and hand it over to it

Reply:
So, without first deserializing?

Reply:
Ideally, the pump deserializes and passes it on to the message handler based on the type.

Maybe better to do a meeting on this?

Reply:
Updated this, can you take a look if it's something you had in mind?